### PR TITLE
Enhance Json Variable Doc with scope example

### LINKED ISSUE #1202: JSON variables: elaborate on scoping
The following statement appears in the [user guide](https://markbind.org/userGuide/reusingContents.html#defining-variables-with-json).
>Variables defined in JSON file will be scoped according to where it is being referenced.

Suggestion: explain this further, possibly with examples.

### PR DESCRIPTION
**Pull request by crphang**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Documentation update

Resolves #1202 

**What is the rationale for this request?**

Elaborate on how JSON variable is scoped.

**What changes did you make? (Give an overview)**

Added documentation.

**Provide some example code that this change will affect:**

N/A

**Is there anything you'd like reviewers to focus on?**

Clarity of documentation.

**Testing instructions:**

N/A
**Proposed commit message: (wrap lines at 72 characters)**

Enhance Json Variable Doc with scope example

### CODE REVIEW

**Comment by crphang** [comment]
Ready for review

File: docs/userGuide/syntax/variables.mbdf [hunk 1]
```diff
@@ -190,7 +190,17 @@ You could also have your variables defined in a JSON file to define multiple var
 }
 ```
 
-Variables defined in JSON file will be scoped according to where it is being referenced.
+Variables defined in JSON file will be scoped according to where it is being referenced. For json variables referenced in `variables.md`
```
Comment (ang-zeyu) [inline docs/userGuide/syntax/variables.mbdf:193]:
```suggestion
Variables defined in a JSON file will be scoped according to where it is being referenced. JSON variables referenced in `variables.md`
```

File: docs/userGuide/syntax/variables.mbdf [hunk 2]
```diff
@@ -190,7 +190,17 @@ You could also have your variables defined in a JSON file to define multiple var
...
+would be globally scoped like other global variables in that file. On the other hand:
```
Comment (ang-zeyu) [inline docs/userGuide/syntax/variables.mbdf:194]:
```suggestion
would be globally scoped like other global variables in that file.
```

**Comment by ang-zeyu** [comment]
Resolved a minor conflict and merged in the suggestions for now, do let me know if the suggestions weren't what you meant!

**Pull request merged**